packages/native/src/components/Form/SelectableList.tsx

```diff
@@ -1,32 +1,48 @@
-import React from "react";
+import React, { ComponentClass, FunctionComponent } from "react";
 import { GestureResponderEvent, TouchableOpacity } from "react-native";
 import styled from "styled-components/native";
 import Flex from "../Layout/Flex";
 import { Text } from "../index";
 
-export type ElementProps<V> = React.PropsWithChildren<{
+type BaseElementProps<V> = {
   first?: boolean;
   selected?: boolean;
   disabled?: boolean;
   value?: V;
   onPress?: ((event: GestureResponderEvent) => void) | undefined;
   Icon?: (props: { size?: number; color?: string }) => React.ReactElement;
-}>;
+};
+
+export type ElementProps<V> = React.PropsWithChildren<
+  BaseElementProps<V> & {
+    /**
+     * A function that will render some content on the right side of the input.
+     */
+    renderRight?:
+      | (
+          | ComponentClass<BaseElementProps<V>>
+          | FunctionComponent<BaseElementProps<V>>
+        )
```

Comment:
Instead of making a union of…

```tsx
| React.ComponentClass<…>
| React.FunctionComponent<…>
```

…you could use the `React.ComponentType<…>` type [which is the exact same](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L82).

Reply:
done